WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -0,0 +1,105 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Crypto.ZeroKnowledge;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class AliceClient
+	{
+		public AliceClient(Guid aliceId, Guid roundId, ArenaClient arenaClient, IEnumerable<Coin> coins, FeeRate feeRate)
+		{
+			AliceId = aliceId;
+			RoundId = roundId;
+			ArenaClient = arenaClient;
+			Coins = coins;
+			FeeRate = feeRate;
+		}
+
+		public Guid AliceId { get; }
+		public Guid RoundId { get; }
+		private ArenaClient ArenaClient { get; }
+		private IEnumerable<Coin> Coins { get; }
+		private FeeRate FeeRate { get; }
+
+		public async Task ConfirmConnectionAsync(TimeSpan confirmInterval, CancellationToken token)
+		{
+			while (!await ConfirmConnectionAsync().ConfigureAwait(false))
+			{
+				await Task.Delay(confirmInterval, token).ConfigureAwait(false);
+			}
+		}
+
+		private async Task<bool> ConfirmConnectionAsync()
+		{
+			var inputWeight = 4 * Constants.P2wpkhInputVirtualSize;
+			var inputRemainingWeights = new[] { (long)ArenaClient.ProtocolMaxWeightPerAlice - inputWeight };
+
+			var amountCredentials = ArenaClient.AmountCredentialClient.Credentials;
+
+			var totalFeeToPay = FeeRate.GetFee(Coins.Sum(c => c.ScriptPubKey.EstimateInputVsize()));
+			var totalAmount = Coins.Sum(coin => coin.Amount);
+
+			if (totalFeeToPay > totalAmount)
```

Comment:
Perhaps this check should be implemented by the `WabiSabiClient`? Because we need to be sure it is impossible to request negative amounts.

Reply:
Maybe. Is it impossible?

Reply:
Yes, It is impossible. The check here is okay, what I say is that it should be also in `WabiSabiClient`.
Btw, if `totalFeeToPay == totalAmount` it should fail too.